### PR TITLE
feat: separate l1blocknumber

### PR DESCRIPTION
**Pull request by smartcontracts**
**Description**
Separates L1 and L2 block numbers. `block.number` will now return the L2 block number. You can access the L1 block number by calling the `OVM_L1BlockNumber` predeploy OR by executing the `L1BLOCKNUMBER` opcode. Predeploy is provided for convenience.

**Metadata**
- Fixes ENG-1364

### CODE REVIEW

File: .github/workflows/sync-tests.yml [hunk 1]
```diff
@@ -34,7 +34,7 @@ jobs:
         working-directory: ./integration-tests
         run: |
           yarn
-          yarn build:integration
```
Comment (smartcontracts) [inline .github/workflows/sync-tests.yml:37]: If desired I can move this diff into a different commit.

File: l2geth/core/evm.go [hunk 2]
```diff
@@ -66,12 +66,13 @@ func NewEVMContext(msg Message, header *types.Header, chain ChainContext, author
...
 			Origin:          msg.From(),
 			Coinbase:        beneficiary,
-			BlockNumber:     msg.L1BlockNumber(),
+			BlockNumber:     new(big.Int).Set(header.Number),
 			Time:            new(big.Int).SetUint64(msg.L1Timestamp()),
 			Difficulty:      new(big.Int).Set(header.Difficulty),
 			GasLimit:        header.GasLimit,
 			GasPrice:        new(big.Int).Set(msg.GasPrice()),
 			L1MessageSender: l1MessageSender,
+			L1BlockNumber:   msg.L1BlockNumber(),
```
Comment (tynes) [inline l2geth/core/evm.go:75]: Doing it this way couples the L1 block number to the txmeta. This is ok for now but we should try to figure out how to eliminate the need for the txmeta. I think the proposed "context tx type" should be able to do this.

**Comment by tynes** [comment]
Generally looks good but no test coverage

**Comment by smartcontracts** [comment]
> Generally looks good but no test coverage

Added integration tests that should cover this

File: integration-tests/contracts/OVMMulticall.sol [hunk 3]
```diff
@@ -45,7 +49,9 @@ contract OVMMulticall {
     }
 
     function getCurrentBlockNumber() public view returns (uint256 blockNumber) {
-        blockNumber = block.number;
+        blockNumber = iOVM_L1BlockNumber(
```
Comment (tynes) [inline integration-tests/contracts/OVMMulticall.sol:52]: Why did you update this but not inside of `aggregate`? I feel like the intended behavior of multicall would be "return the L2 blocknumber"

**Comment by tynes** [comment]
> > Generally looks good but no test coverage
> 
> Added integration tests that should cover this

It would be a lot better to have a unit test in geth that tests the functionality rather than an integration test. The integration test approach touches way more codepaths instead of testing just the bare minimal thing that we want to test here.

**Review by tynes — APPROVED** [review]

**Pull request merged**